tcwebhooks-core/src/main/java/webhook/teamcity/WebHookContentBuilder.java

```diff
@@ -229,6 +230,15 @@ public WebHook buildWebHookContent(WebHook wh, WebHookConfig whc, SBuild sBuild,
 				wh.checkFilters(builder);
 				wh.resolveHeaders(builder);
 			}
+		} else if (state.equals(BuildStateEnum.BUILD_TAGGED)) {
+			wh.setEnabledForBuildState(state, isOverrideEnabled || (whc.isEnabledForBuildType(sBuild.getBuildType()) && wh.getBuildStates().enabled(state)));
+			if (wh.isEnabled()){
+				templateForThisBuild = findTemplateForState(sBuild, state, whc.getPayloadTemplate());
+				wh.setPayload(payloadFormat.buildPromotionTagsChanged(sBuild, getPreviousNonPersonalBuild(wh, sBuild), mergeParameters(whc.getParams(),sBuild, getPreferredDateFormat(templateForThisBuild)), whc.getEnabledTemplates(), templateForThisBuild));
```

Comment:
```suggestion
				wh.setPayload(payloadFormat.buildTagsChanged(sBuild, getPreviousNonPersonalBuild(wh, sBuild), mergeParameters(whc.getParams(),sBuild, getPreferredDateFormat(templateForThisBuild)), whc.getEnabledTemplates(), templateForThisBuild));
```

Reply:
I don't think we need `getPreviousNonPersonalBuild(wh, sBuild)` here. 
It is used to find the previous build so that we can determine if the build status (failed/successful) has changed, so that we can generate the pseudo states of "fixed" and "broken".

For tags I don't believe it makes sense. Tags typically can be applied regardless of the previous build result.